### DISCUSSION TITLE
Export ADFS Configuration issue

### DISCUSSION THREAD

**Discussion by RangerJavelin** [Azure/SimuLand#16] [Q&A]
I perform the AAD Internal PowerShell in WORKSTATION machine with domain admin account `pgustavo`
` $settings = Export-AADIntADFSConfiguration -Hash $NTHash -SID $ObjectSid -Server $ADFSServer`
but it shows me an error

`PS C:\Windows\system32> $settings = Export-AADIntADFSConfiguration -Hash $NTHash -SID $ObjectSid -Server $ADFSServer
Cannot index into a null array.`
At C:\Program Files\WindowsPowerShell\Modules\AADInternals\0.4.8\ADFS_utils.ps1:568 char:9
+         $krb_response =  Convert-B64ToByteArray -B64 $RSTR.Envelope.B ...
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : NullArray
 
Cannot index into a null array.
At C:\Program Files\WindowsPowerShell\Modules\AADInternals\0.4.8\ADFS_utils.ps1:569 char:9
+         $proofToken =    Convert-B64ToByteArray -B64 $RSTR.Envelope.B ...
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : NullArray
 
Cannot index into a null array.
At C:\Program Files\WindowsPowerShell\Modules\AADInternals\0.4.8\ADFS_utils.ps1:570 char:9
+         $keyIdentifier = [guid](($RSTR.Envelope.Body.RequestSecurityT ...
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : NullArray
 
Cannot index into a null array.
At C:\Program Files\WindowsPowerShell\Modules\AADInternals\0.4.8\ADFS_utils.ps1:571 char:9
+         $context =       $RSTR.Envelope.Body.RequestSecurityTokenResp ...
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : NullArray
 
Parse-Asn1 : Cannot bind argument to parameter 'Data' because it is null.
At C:\Program Files\WindowsPowerShell\Modules\AADInternals\0.4.8\ADFS_utils.ps1:574 char:36
+         $ticket = Parse-Asn1 -Data $krb_response
+                                    ~~~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (:) [Parse-Asn1], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorNullNotAllowed,Parse-Asn1
 
Cannot index into a null array.
At C:\Program Files\WindowsPowerShell\Modules\AADInternals\0.4.8\ADFS_utils.ps1:577 char:9
+         $encKRB = $ticket.Data.Data[2].Data.Data
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : NullArray
 
Parse-Asn1 : Cannot bind argument to parameter 'Data' because it is null.
At C:\Program Files\WindowsPowerShell\Modules\AADInternals\0.4.8\ADFS_utils.ps1:578 char:36
+         $encKRB = Parse-Asn1 -Data $encKRB
+                                    ~~~~~~~
    + CategoryInfo          : InvalidData: (:) [Parse-Asn1], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorNullNotAllowed,Parse-Asn1
 
Cannot index into a null array.
At C:\Program Files\WindowsPowerShell\Modules\AADInternals\0.4.8\ADFS_utils.ps1:579 char:9
+         $encKRB = $encKRB.Data[2].Data.Data[2].Data.Data[1].Data.Data
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : NullArray
 
Decrypt-Kerberos : Cannot bind argument to parameter 'Data' because it is null.
At C:\Program Files\WindowsPowerShell\Modules\AADInternals\0.4.8\ADFS_utils.ps1:582 char:42
+         $decKRB = Decrypt-Kerberos -Data $encKRB -Key $Key -Type APRe ...
+                                          ~~~~~~~
    + CategoryInfo          : InvalidData: (:) [Decrypt-Kerberos], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorNullNotAllowed,Decrypt-Kerberos
 
Parse-Asn1 : Cannot bind argument to parameter 'Data' because it is null.
At C:\Program Files\WindowsPowerShell\Modules\AADInternals\0.4.8\ADFS_utils.ps1:583 char:36
+         $decKRB = Parse-Asn1 -Data $decKRB
+                                    ~~~~~~~
    + CategoryInfo          : InvalidData: (:) [Parse-Asn1], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorNullNotAllowed,Parse-Asn1
 
Cannot index into a null array.
At C:\Program Files\WindowsPowerShell\Modules\AADInternals\0.4.8\ADFS_utils.ps1:586 char:9
+         $SequenceNumber = $decKRB.Data.Data[3].Data.Data
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : NullArray
 
Cannot index into a null array.
At C:\Program Files\WindowsPowerShell\Modules\AADInternals\0.4.8\ADFS_utils.ps1:587 char:9
+         $SubKey =         $decKRB.Data.Data[2].Data.Data[1].Data.Data
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : NullArray
 
Cannot index into a null array.
At C:\Program Files\WindowsPowerShell\Modules\AADInternals\0.4.8\ADFS_utils.ps1:588 char:9
+         $EncryptionType = $decKRB.Data.Data[2].Data.Data[0].Data.Data
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : NullArray
 
Convert-ByteArrayToB64 : Cannot bind argument to parameter 'Bytes' because it is null.
At C:\Program Files\WindowsPowerShell\Modules\AADInternals\0.4.8\ADFS_utils.ps1:590 char:71
+ ... ite-Verbose "Subkey:        $(Convert-ByteArrayToB64 -Bytes $subKey)"
+                                                                 ~~~~~~~
    + CategoryInfo          : InvalidData: (:) [Convert-ByteArrayToB64], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorNullNotAllowed,Convert-ByteArrayToB64
 
Parse-GSS_Wrap : Cannot bind argument to parameter 'Data' because it is null.
At C:\Program Files\WindowsPowerShell\Modules\AADInternals\0.4.8\ADFS_utils.ps1:594 char:68
+ ... Key = (Parse-GSS_Wrap -Type $EncryptionType -Data $proofToken -Sessio ...
+                                                       ~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (:) [Parse-GSS_Wrap], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorNullNotAllowed,Parse-GSS_Wrap
 
Convert-ByteArrayToB64 : Cannot bind argument to parameter 'Bytes' because it is null.
At C:\Program Files\WindowsPowerShell\Modules\AADInternals\0.4.8\ADFS_utils.ps1:596 char:71
+ ... erbose "Security key:  $(Convert-ByteArrayToB64 -Bytes $securityKey)"
+                                                            ~~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (:) [Convert-ByteArrayToB64], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorNullNotAllowed,Convert-ByteArrayToB64
 
Create-SCTEnvelope : Cannot bind argument to parameter 'Key' because it is null.
At C:\Program Files\WindowsPowerShell\Modules\AADInternals\0.4.8\ADFS.ps1:279 char:54
+ ...            $envelope =      Create-SCTEnvelope -Key $RSTR.Key -Client ...
+                                                         ~~~~~~~~~
    + CategoryInfo          : InvalidData: (:) [Create-SCTEnvelope], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorNullNotAllowed,Create-SCTEnvelope
 
The request for security token has invalid or malformed elements.
At C:\Program Files\WindowsPowerShell\Modules\AADInternals\0.4.8\ADFS_utils.ps1:691 char:13
+             throw $reason
+             ~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (The request for...ormed elements.:String) [], RuntimeException
    + FullyQualifiedErrorId : The request for security token has invalid or malformed elements.

**Comment by Cyb3rWard0g** [comment]
Hello @RangerJavelin ! Sorry for the late response. I have to test that just in case. I wonder if maybe that's something that we can ask also the creator of the tool: https://github.com/Gerenios/AADInternals.

Would you mind updating it to the latest version?

**Comment by chrisdavidfriedrich-ctrl** [comment]
Hello @RangerJavelin how can I contact you for a service? thank you
